### PR TITLE
Sheriff's Authority Implementation

### PR DESCRIPTION
**Pull request by Ignose**
This requires my Mafia PR before it can be used.

If already whitelisted to home clan, seeks a BAFH whitelist. 
   If you already have one, joins BAFH, grabs sheriff parts, and uses authority as a new free kill resource.
Returns to original clan

Adds cases for the +com and -com skills from photobooth to several quests, following Apriling logic. An improvement may be one "tuneable" function.

### CODE REVIEW

File: src/engine/resources.ts [hunk 1]
```diff
@@ -735,6 +742,21 @@ export function tryPlayApriling(modifier: string): void {
   }
 }
 
+export function photoboothEffect(modifier: string): void {
```
Comment (Kasekopf) [inline src/engine/resources.ts]: We shouldn't do this in the same way as Apriling, since we can't switch back-and-forth as often. Probably better to stick to -combat and treat it similarly to submarine sprints. https://github.com/Kasekopf/looprobot/blob/d611e5144b3cca42b0a8059141d2661cc905bad0/src/engine/moods.ts#L78

File: src/tasks/misc.ts [hunk 2]
```diff
@@ -1133,6 +1136,32 @@ export const MiscQuest: Quest = {
       freeaction: true,
       limit: { tries: 2 },
     },
+    {
+      name: "Clan Photo Booth Free Kill",
+      after: [],
+      priority: () => Priorities.Free,
+      ready: () =>
+          Clan.getWhitelisted().find(
```
Comment (Kasekopf) [inline src/tasks/misc.ts]: Don't make web calls in a `ready` call. Probably just call this once at the start of the script.

File: src/tasks/misc.ts [hunk 3]
```diff
@@ -1133,6 +1136,32 @@ export const MiscQuest: Quest = {
...
+          && get("_photoBoothEquipment", 0) === 0 ,
+      completed: () => (have($item`Sheriff moustache`) && have($item`Sheriff badge`) && have($item`Sheriff pistol`)) || get("_photoBoothEquipment", 0) >= 3,
+      do: (): void => {
+        const currentClan = getClanName();
+        if(
+        (Clan.getWhitelisted().find(
+        (c) => c.name === "Bonus Adventures from Hell",
+        ) === undefined))
+          chatPrivate("bafhbot", "whitelist bafh")
+        if(currentClan !== "Bonus Adventures from Hell") Clan.join("Bonus Adventures from Hell")
```
Comment (Kasekopf) [inline src/tasks/misc.ts]: Libram has `withClan` which should help with this.

File: src/tasks/misc.ts [hunk 4]
```diff
@@ -1133,6 +1136,32 @@ export const MiscQuest: Quest = {
...
+          ) !== undefined
+          && get("_photoBoothEquipment", 0) === 0 ,
+      completed: () => (have($item`Sheriff moustache`) && have($item`Sheriff badge`) && have($item`Sheriff pistol`)) || get("_photoBoothEquipment", 0) >= 3,
+      do: (): void => {
+        const currentClan = getClanName();
+        if(
+        (Clan.getWhitelisted().find(
+        (c) => c.name === "Bonus Adventures from Hell",
+        ) === undefined))
+          chatPrivate("bafhbot", "whitelist bafh")
```
Comment (Kasekopf) [inline src/tasks/misc.ts]: Don't we need to wait and see if the bot responds? I would almost say this is out of scope, and we can just assume the user has a BAFH whitelist, or skip the photobooth.

Comment (libraryaddict) [reply src/tasks/misc.ts]: I'd say definitely don't make them apply for the clan, at most just log a message explaining why they can't join the clan.

Imagine running garbo and finding out it automatically puts all your bags of garbage in store at 210 meat.
Also, not everyone has a whitelist to their own clan so you could be effectively kicking them out of their clan if they're not whitelisted.

Comment (Ignose) [reply src/tasks/misc.ts]: I check that in the ready condition; you need to be whitelisted to both clans!

File: src/tasks/misc.ts [hunk 5]
```diff
@@ -86,6 +88,14 @@ import { ROUTE_WAIT_TO_NCFORCE } from "../route";
 
 const meatBuffer = 1000;
 
+const sheriffReady = Clan.getWhitelisted().find(
```
Comment (Kasekopf) [inline src/tasks/misc.ts]: Could you pull out the two Clan.getWhitelisted() calls into a single variable as well? So we only make one web request.

File: src/engine/moods.ts [hunk 6]
```diff
@@ -71,11 +71,19 @@ function getRelevantEffects(): { [modifier: string]: Effect[] } {
...
-    result["-combat"].push($effect`Silent Running`);
+    result["-combat"].push($effect`Wild and Westy!`);
+
+    if (
+      have($item`Clan VIP Lounge key`) &&
+      !underStandard() &&
+      (!get("_olympicSwimmingPool") || have($effect`Silent Running`)) &&
+      ((have($item`ninja rope`) && have($item`ninja carabiner`) && have($item`ninja crampons`)) ||
+        step("questL08Trapper") >= 3)
+    )
```
Comment (Kasekopf) [inline src/engine/moods.ts:86]: You accidentally deleted the `result["-combat"].push($effect`Silent Running`);` line.

**Comment by Kasekopf** [comment]
Thanks for your work on this PR!

In the future, could you run `yarn lint:fix` to set the formatting of your changes when making commits/PRs? Otherwise there ends up being formatting changes needed in a [later commit](https://github.com/Kasekopf/looprobot/commit/a1497f88fb1031691b35261f423af528402d95bb).
